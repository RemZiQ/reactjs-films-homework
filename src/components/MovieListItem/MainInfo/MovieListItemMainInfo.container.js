import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { genres } from '../../../modules/module_Search/selectors';
import MainInfo from './MovieListItemMainInfo';


const mapStateToProps = createStructuredSelector({ storeGenres: genres });


export default connect(
  mapStateToProps, null,
)(MainInfo);
