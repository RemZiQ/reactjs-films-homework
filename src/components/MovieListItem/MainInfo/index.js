export { default } from './MovieListItemMainInfo.container';
