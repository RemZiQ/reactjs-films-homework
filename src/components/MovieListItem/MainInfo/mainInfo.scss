.movieList__item__mainInfo {
  position: relative;
}

.movieList__item_genresList li {
  display: inline-block;
  margin-right: 5px;
  color: #47c3f0;
  font-size: 18px;
  letter-spacing: 1px;
  line-height: 33px;
  font-family: "Helvetica Neue", sans-serif;
  font-weight: 500;
}

.movieList__item_mark {
  position: absolute;
  display: inline-block;
  box-sizing: border-box;
  width: 60px;
  height: 36px;
  text-align: center;
  line-height: 36px;
  border: 2px solid #47c3f0;
  border-radius: 5px;
  font-weight: bold;
  color: #47c3f0;
}

.movieList__item_title {
  display: inline-block;
  width: 70%;
  margin-right: 10%;
  font-size: 28px;
  letter-spacing: 1px;
  line-height: 44px;
  font-family: "Helvetica Neue", sans-serif;
  font-weight: bold;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}
