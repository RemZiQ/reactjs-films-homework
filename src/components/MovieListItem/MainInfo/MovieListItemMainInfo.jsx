import React from 'react';
import propTypes from 'prop-types';
import { connect } from 'react-redux';

import './mainInfo.scss';


const MainInfo = ({
  title, mark, genres, storeGenres,
}) => {
  const mapGenres = storeGenres;
  let listOfGenres = '';
  if (storeGenres.length) {
    listOfGenres = genres.map(item => (
      <li key={item}>{mapGenres.filter(elem => elem.id === item)[0].name}</li>
    ));
  }
  return (
    <div className="movieList__item__overview__mainInfo">
      <h2 className="movieList__item_title">{title}</h2>
      <div className="movieList__item_mark">{mark}</div>
      <ul className="movieList__item_genresList">{listOfGenres}</ul>
    </div>
  );
};


MainInfo.propTypes = {
  title: propTypes.string,
  mark: propTypes.number,
  genres: propTypes.arrayOf(propTypes.number).isRequired,
  storeGenres: propTypes.arrayOf(propTypes.object).isRequired,
};

MainInfo.defaultProps = {
  title: 'Sorry, no title here',
  mark: 'Sorry, no mark here',
};


const mapStateToProps = store => ({ storeGenres: store.genres });

export default connect(
  mapStateToProps, null,
)(MainInfo);
