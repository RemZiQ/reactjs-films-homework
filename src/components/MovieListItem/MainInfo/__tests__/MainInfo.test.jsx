import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import MainInfo from '../MovieListItemMainInfo';


it('MainInfo renders well with storeGenres', () => {
  const props = {
    genres: [1, 2],
    storeGenres: [{ id: 1, name: 'someGenre' }, { id: 2, name: 'someGenre2' }],
  };
  const renderer = new ShallowRenderer();
  const result = renderer.render(<MainInfo {...props} />);
  expect(result).toMatchSnapshot();
});

it('MainInfo renders well without storeGenres', () => {
  const props = {
    genres: [1, 2],
    storeGenres: [],
  };
  const renderer = new ShallowRenderer();
  const result = renderer.render(<MainInfo {...props} />);
  expect(result).toMatchSnapshot();
});
