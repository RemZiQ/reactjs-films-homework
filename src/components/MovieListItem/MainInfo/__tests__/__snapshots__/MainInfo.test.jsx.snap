// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainInfo renders well with storeGenres 1`] = `
<div
  className="movieList__item__overview__mainInfo"
>
  <h2
    className="movieList__item_title"
  >
    Sorry, no title here
  </h2>
  <div
    className="movieList__item_mark"
  >
    Sorry, no mark here
  </div>
  <ul
    className="movieList__item_genresList"
  >
    <li>
      someGenre
    </li>
    <li>
      someGenre2
    </li>
  </ul>
</div>
`;

exports[`MainInfo renders well without storeGenres 1`] = `
<div
  className="movieList__item__overview__mainInfo"
>
  <h2
    className="movieList__item_title"
  >
    Sorry, no title here
  </h2>
  <div
    className="movieList__item_mark"
  >
    Sorry, no mark here
  </div>
  <ul
    className="movieList__item_genresList"
  >
    
  </ul>
</div>
`;
